src/common/com/intellij/plugins/haxe/model/HaxeSourceRootModel.java

```diff
@@ -50,6 +51,9 @@ public boolean contains(PsiFileSystemItem file) {
     if (virtualFile == null) {
       virtualFile = file.getOriginalElement().getContainingFile().getVirtualFile();
     }
+    if (virtualFile == null) {
+      virtualFile = file.getUserData(IndexingDataKeys.VIRTUAL_FILE);
```

Comment:
Well, hell.  What a workaround.  But since JetBrains does this for their own indexing, it must be a valid way to go about it.  Good job finding that.

Reply:
I've spent two days in deep debugging to write this three lines 😅

Reply:
Some days it goes like that...  It's good to get the problem fixed, though.